main.go

```diff
@@ -14,7 +15,386 @@
+			registrysendfd.NewNetworkServiceEndpointRegistryClient(),
+		),
+	)
+	nse := getNseEndpoint(config, listenOn)
+	nse, err = nseRegistryClient.Register(ctx, nse)
+	logrus.Infof("nse: %+v", nse)
+
+	if err != nil {
+		log.FromContext(ctx).Fatalf("unable to register nse %+v", err)
+	}
+
+	// ********************************************************************************
+	log.FromContext(ctx).Infof("startup completed in %v", time.Since(starttime))
+	// ********************************************************************************
+
+	// wait for server to exit
+	<-ctx.Done()
+}
+
+func getNseEndpoint(config *Config, listenOn fmt.Stringer) *registryapi.NetworkServiceEndpoint {
+	nse := &registryapi.NetworkServiceEndpoint{
+		Name:                 config.Name,
+		NetworkServiceNames:  config.ServiceNames,
+		NetworkServiceLabels: make(map[string]*registryapi.NetworkServiceLabels),
+		Url:                  listenOn.String(),
+	}
+	for _, serviceName := range config.ServiceNames {
+		nse.NetworkServiceLabels[serviceName] = &registryapi.NetworkServiceLabels{Labels: config.Labels}
+	}
+	return nse
+}
+
+func getSriovTokenServerChainElement(ctx context.Context) (tokenServer networkservice.NetworkServiceServer) {
+	sriovTokens := tokens.FromEnv(os.Environ())
+	switch len(sriovTokens) {
+	case 0:
+		tokenServer = null.NewServer()
+	case 1:
+		var tokenKey string
+		for tokenKey = range sriovTokens {
+			break
+		}
+		tokenServer = token.NewServer(tokenKey)
+	default:
+		log.FromContext(ctx).Fatalf("endpoint must be configured with none or only one sriov resource")
+	}
+	return
+}
+
+func exitOnErr(ctx context.Context, cancel context.CancelFunc, errCh <-chan error) {
+	// If we already have an error, log it and exit
+	select {
+	case err := <-errCh:
+		log.FromContext(ctx).Fatal(err)
+	default:
+	}
+	// Otherwise wait for an error in the background to log and cancel
+	go func(ctx context.Context, errCh <-chan error) {
+		err := <-errCh
+		log.FromContext(ctx).Error(err)
+		cancel()
+	}(ctx, errCh)
+}
+
+// newPolicyRoutesGetter - watches the config file and dynamically updates policy routes.
+func newPolicyRoutesGetter(ctx context.Context, configPath string) *policyRoutesGetter {
+	p := &policyRoutesGetter{
+		ctx: ctx,
+	}
+	p.policyRoutes.Store([]*networkservice.PolicyRoute{})
+	updatePrefixes := func(bytes []byte) {
+		if bytes == nil {
+			p.policyRoutes.Store([]*networkservice.PolicyRoute{})
+		}
+		var source []*networkservice.PolicyRoute
+		err := yaml.Unmarshal(bytes, &source)
+		if err != nil {
+			log.FromContext(ctx).Errorf("Cannot unmarshal policies, err: %v", err.Error())
+			return
+		}
+		p.policyRoutes.Store(source)
+	}
+	updateCh := fs.WatchFile(p.ctx, configPath)
+	updatePrefixes(<-updateCh)
+	go func() {
+		for {
+			select {
+			case <-p.ctx.Done():
+				return
+			case update := <-updateCh:
+				updatePrefixes(update)
+			}
+		}
+	}()
+
+	return p
+}
+
+func (p *policyRoutesGetter) Get() []*networkservice.PolicyRoute {
```

Comment:
This whole function is probably unneeded in this NSE

Reply:
Done.